src/main/java/teammates/common/datatransfer/FeedbackQuestionAttributes.java

```diff
@@ -601,4 +607,75 @@ public String getQuestionAdditionalInfoHtml() {
+        for (FeedbackPath feedbackPath : feedbackPaths) {
+            feedbackPathsAttributesList.add(new FeedbackPathAttributes(feedbackPath));
+        }
+        return feedbackPathsAttributesList;
+    }
+    
+    public List<FeedbackPath> getFeedbackPaths(List<FeedbackPathAttributes> feedbackPathsAttributesList) {
+        List<FeedbackPath> feedbackPaths = new ArrayList<FeedbackPath>();
+        if (feedbackPathsAttributesList != null) {
+            for (FeedbackPathAttributes feedbackPath : feedbackPathsAttributesList) {
+                feedbackPaths.add(feedbackPath.toEntity());
+            }
+        }
+        
+        return feedbackPaths;
+    }
+    
+    public List<FeedbackPath> getFeedbackPaths() {
+        return getFeedbackPaths(feedbackPathAttributesList);
+    }
+    
+    /** 
+     * Verifies whether the given student is a giver in the question's feedback path attributes list
+     */
+    public boolean hasStudentAsGiverInFeedbackPathAttributesList(StudentAttributes student) {
+        if (giverType == FeedbackParticipantType.CUSTOM && recipientType == FeedbackParticipantType.CUSTOM) {
+            for (FeedbackPathAttributes feedbackPathAttributes : feedbackPathAttributesList) {
+                String[] giver = feedbackPathAttributes.getGiver().split(" ");
+                if (giver[0].equals(student.getTeam())
+                        || giver[0].equals(student.getEmail()) && "(Student)".equals(giver[1])) {
+                    return true;
+                }
+            }
+        }
+        return false;
+    }
+    
+    /** 
+     * Verifies whether the given instructor is a giver in the question's feedback path attributes list
+     */
+    public boolean hasInstructorAsGiverInFeedbackPathAttributesList(String instructorEmail) {
+        if (giverType == FeedbackParticipantType.CUSTOM && recipientType == FeedbackParticipantType.CUSTOM) {
+            for (FeedbackPathAttributes feedbackPathAttributes : feedbackPathAttributesList) {
+                String[] giver = feedbackPathAttributes.getGiver().split(" ");
+                if (giver[0].equals(instructorEmail) && "(Instructor)".equals(giver[1])) {
+                    return true;
+                }
+            }
+        }
+        return false;
+    }
+    
+    public static List<FeedbackPathAttributes> getFeedbackPathAttributesListFromSpreadsheetData(
+            String courseId, String customFeedbackPathsSpreadsheetData) {
+        Gson gson = new Gson();
+        TypeToken<List<List<String>>> token = new TypeToken<List<List<String>>>(){};
+        List<List<String>> customFeedbackPaths =
+                gson.fromJson(customFeedbackPathsSpreadsheetData, token.getType());
+        List<FeedbackPathAttributes> feedbackPathAttributesList = new ArrayList<FeedbackPathAttributes>();
+        for (List<String> feedbackPath : customFeedbackPaths) {
+            if (!feedbackPath.contains("") && !feedbackPath.contains(null)) {
```

Comment:
Not sure how to make it possible, but it will be nice if we can remove the check for `null`s in lists